src/stateMachines/OID4VCIMachine.tsx

```diff
@@ -0,0 +1,534 @@
+          onDone: {
+            target: OID4VCIMachineStates.retrievingContact,
+            actions: assign({supportedCredentials: (_ctx: OID4VCIMachineContext, _event: DoneInvokeEvent<any>) => _event.data}),
+            // TODO add condition that we have at least 1 vc in the selection
+          },
+          onError: {
+            target: OID4VCIMachineStates.showingError,
+            actions: assign({
+              error: (_ctx: OID4VCIMachineContext, _event: DoneInvokeEvent<Error>): ErrorDetails => ({
+                title: 'Credential selection', // TODO translate
+                message: _event.data.message,
+              }),
+            }),
+          },
+        },
+      },
+      [OID4VCIMachineStates.retrievingContact]: {
+        id: OID4VCIMachineStates.retrievingContact,
+        invoke: {
+          src: OID4VCIMachineServices.retrieveContact,
+          onDone: {
+            target: OID4VCIMachineStates.transitioningFromSetup,
+            actions: assign({contact: (_ctx: OID4VCIMachineContext, _event: DoneInvokeEvent<any>) => _event.data}), // TODO any and all others
+          },
+          onError: {
+            target: OID4VCIMachineStates.showingError,
+            actions: assign({
+              error: (_ctx: OID4VCIMachineContext, _event: DoneInvokeEvent<Error>): ErrorDetails => ({
+                title: 'retrieve contact', // TODO translate
+                message: _event.data.message,
+              }),
+            }),
+          },
+        },
+      },
+      [OID4VCIMachineStates.transitioningFromSetup]: {
+        id: OID4VCIMachineStates.transitioningFromSetup,
+        always: [
+          {
+            target: OID4VCIMachineStates.addingContact,
+            cond: OID4VCIMachineGuards.hasNotContactGuard,
+          },
+          {
+            target: OID4VCIMachineStates.selectingCredentials,
+            cond: OID4VCIMachineGuards.selectCredentialsGuard,
+          },
+          {
+            target: OID4VCIMachineStates.authenticating,
+            cond: OID4VCIMachineGuards.authenticationGuard,
+          },
+          {
+            target: OID4VCIMachineStates.retrievingCredentials,
+          },
+        ],
+      },
+      [OID4VCIMachineStates.addingContact]: {
+        id: OID4VCIMachineStates.addingContact,
+        on: {
+          [OID4VCIMachineEvents.SET_CONTACT_CONSENT]: {
+            actions: assign({hasContactConsent: (_ctx: OID4VCIMachineContext, _event: ContactConsentEvent) => _event.data}),
+          },
+          [OID4VCIMachineEvents.SET_CONTACT_ALIAS]: {
+            actions: assign({contactAlias: (_ctx: OID4VCIMachineContext, _event: ContactAliasEvent) => _event.data}),
+          },
+          [OID4VCIMachineEvents.CREATE_CONTACT]: {
+            cond: OID4VCIMachineGuards.createContactGuard,
+            actions: [assign({contact: (_ctx: OID4VCIMachineContext, _event: CreateContactEvent) => _event.data}), send(OID4VCIMachineEvents.NEXT)],
```

Comment:
I am doing this so that i can guard against having a contact to be able to continue. i cannot (or at least i cannot find a nice way to do it) assign a value and check against it in the same event. To me its seems like something common to assign something in the context and to check it as well, so unless you have an idea on how i should solve it i need to do some reading in the documentation

Reply:
i placed a todo comment for now